tiatoolbox/wsicore/wsireader.py

```diff
@@ -362,6 +373,15 @@ def read_rect(self, location, size, resolution=0, units="level"):
                 Supported units are: microns per pixel (mpp), objective
                 power (power), pyramid / resolution level (level),
                 pixels per baseline pixel (baseline).
+            interpolation (str): Method to use when resampling the output
+                image. Possible values are "linear", "cubic", "lanczos",
+                "area", and "optimse". Defaults to 'optimise' which
+                will use cubic interpolation for upscaling and area
+                interpolation for downscaling to avoid moiré patterns.
+            pad_mode (str): Method to use when padding at the edges of the
+                image. Defaults to 'constant'. See :func:`numpy.pad` for
+                available modes.
+            **kwargs (dict): Extra key-word arguments.
```

Comment:
Extra keyword arguments for which function? To a numpy function? Can we add some reference?

Reply:
This is intended for experimental / class specific kwargs. Not sure how best to document with docstrings without duplicating this very large docstring eight times. Something we shold discuss further. Currently, this is only used for VirtualWSIReader and passes them on to the sub_pixel_read function.

Reply:
@shaneahmed @vqdang  I've added some detail here (including links) which might help clarify. https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/ddfe109863815373ffbe2396fdfe01e14468b7e7

Reply:
@shaneahmed @vqdang  I've added some detail here (including links) which might help clarify. https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/ddfe109863815373ffbe2396fdfe01e14468b7e7

Reply:
Thanks, please link with :obj:`VirtualWSIReader` in one of docstrings.